movies/src/tests/tests.hydra

```diff
@@ -0,0 +1,34 @@
+PREFIX schema: <http://schema.org/>
+PREFIX hydra: <http://www.w3.org/ns/hydra/core#>
+PREFIX mov: <https://hydra-movies.herokuapp.com/doc#>
+
+With Class schema:EntryPoint {
+    Expect Property hydra:collection {
+        Expect Type hydra:Collection
```

Comment:
This seems unclear. I assume this assertion is about an object of the `hydra:collection` relation, but first impression is that the assertion is about a relation itself. The `hydra:Collection` is somehow unnecessary - the hydra vocab states clearly that it's values are of that type.

Reply:
> This seems unclear. I assume this assertion is about an object of the hydra:collection relation

Yes, the idea is that every time a block is opened, the test traverses a graph edge (`Property`) or follows a `Link` to another resource. The statements then apply to the resource found on the other end of that relation.

> but first impression is that the assertion is about a relation itself

Could you explain how you arrive at such conclusion?

> the hydra vocab states clearly that it's values are of that type.

I see where you're coming from. But, that would require applying inferences to the resource graph. Here it would add explicit `<> a hydra:Collection` where it's only implicit. Is that something that Heracles.ts is doing to representations?

Still, I agree that it may be unnecessary for the test scenario either way.

Reply:
>> but first impression is that the assertion is about a relation itself

> Could you explain how you arrive at such conclusion?

As I wrote - it was an impression, possibly subjective one. Maybe it's from OOP, where class is described with its members (i.e. properties) and its members are described by their body or other metadata.

> But, that would require applying inferences to the resource graph. Here it would add explicit > `<> a hydra:Collection` where it's only implicit. Is that something that Heracles.ts is doing to > representations?

Yes and no. Heracles.ts does not use raw RDF as it's internal data model. Still, it has some infrastructure that tries to match predicates with simple inference mechanism that checks range/domain provided by the hydra vocabulary RDF representation. There is a test that has a JSON-LD input that provides two `hydra:collection` links and one of them is untyped, yet still discovered as a collection. Same behavior is reflected in the Heracles.net that I'm writing in the meanwhile.